src/main/java/com/redhat/devtools/intellij/quarkus/lsp4ij/console/explorer/LanguageServerExplorerLifecycleListener.java

```diff
@@ -0,0 +1,171 @@
+import org.eclipse.lsp4j.jsonrpc.messages.Message;
+import org.eclipse.lsp4j.jsonrpc.messages.NotificationMessage;
+import org.eclipse.lsp4j.jsonrpc.messages.RequestMessage;
+import org.eclipse.lsp4j.jsonrpc.messages.ResponseMessage;
+
+import javax.swing.tree.DefaultMutableTreeNode;
+import javax.swing.tree.TreePath;
+
+import static com.redhat.devtools.intellij.quarkus.lsp4ij.utils.UIHelper.executeInUI;
+
+public class LanguageServerExplorerLifecycleListener implements LanguageServerLifecycleListener {
+
+    private final LanguageServerExplorer explorer;
+
+    public LanguageServerExplorerLifecycleListener(LanguageServerExplorer explorer) {
+        this.explorer = explorer;
+    }
+
+    @Override
+    public void handleStartingProcess(LanguageServerWrapper languageServer) {
+        executeInUI(() -> {
+            if (!isAdaptedFor(languageServer)) {
+                return;
+            }
+            LanguageServerProcessTreeNode processTreeNode = findLanguageServerProcessTreeNode(languageServer);
+            processTreeNode.setServerStatus(ServerStatus.startingProcess);
+            explorer.selectAndExpand(processTreeNode);
+        });
+    }
+
+    @Override
+    public void handleStartedProcess(LanguageServerWrapper languageServer, Throwable exception) {
+        executeInUI(() -> {
+            if (!isAdaptedFor(languageServer)) {
+                return;
+            }
+            LanguageServerProcessTreeNode processTreeNode = findLanguageServerProcessTreeNode(languageServer);
+            processTreeNode.setServerStatus(ServerStatus.startedProcess);
+        });
+    }
+
+    @Override
+    public void handleStartedLanguageServer(LanguageServerWrapper languageServer, Throwable exception) {
+        executeInUI(() -> {
+            if (!isAdaptedFor(languageServer)) {
+                return;
+            }
+            LanguageServerProcessTreeNode processTreeNode = findLanguageServerProcessTreeNode(languageServer);
+            processTreeNode.setServerStatus(ServerStatus.started);
+        });
+    }
+
+    @Override
+    public void handleLSPMessage(Message message, LanguageServerWrapper languageServer) {
+        if (!isAdaptedFor(languageServer)) {
+            return;
+        }
+        ServerTrace serverTrace = getServerTrace(languageServer.serverDefinition.id);
+        if (serverTrace == ServerTrace.off) {
+            return;
+        }
+
+        StringBuilder formattedMessage = new StringBuilder("[Trace]");
+        if (message instanceof RequestMessage) {
+            // [Trace - 12:27:33 AM] Sending request 'initialize - (0)'.
+            //  Params: {
+            formattedMessage.append(" Sending request '" + ((RequestMessage) message).getMethod() + " - (" + ((RequestMessage) message).getId() + ")'.");
+        } else if (message instanceof ResponseMessage) {
+            // [Trace - 12:27:35 AM] Received response 'initialize - (0)' in 1921ms.

```

Comment:
```java
String id = ((ResponseMessage) message).getId();
String method = methods.remove(id);
formattedMessage.append(" Received response '").append(method == null?"<unknown>":method).append(" - (").append(id).append(")'.");
```

Reply:
this fixes responses missing the original method call name:
<img width="489" alt="Screenshot 2023-05-16 at 17 06 20" src="https://github.com/redhat-developer/intellij-quarkus/assets/148698/ed6197e4-1d10-49ea-8707-14d4e41f493b">

Reply:
OkI applied your suggestion and improve it to show the date after the Trace and the elapsed time and I take care of multiple language servers:

![image](https://github.com/redhat-developer/intellij-quarkus/assets/1932211/ac4923ca-93f5-443d-b0cb-ee9210eda8ed)